src/ducks/txHistory.ts

```diff
@@ -0,0 +1,56 @@
+import { createSlice } from "@reduxjs/toolkit";
+import { DataProvider, Types } from "@stellar/wallet-sdk";
+import StellarSdk from "stellar-sdk";
+import { AppDispatch } from "App";
+
+interface InitialTxHistoryState {
+  data: Array<Types.Payment>;
+  error?: string;
+  loading: boolean;
+}
+
+const initialTxHistoryState: InitialTxHistoryState = {
+  data: [],
+  error: undefined,
+  loading: false,
+};
+
+export const txHistorySlice = createSlice({
+  name: "txHistory",
+  initialState: initialTxHistoryState,
+  reducers: {
+    txHistoryPending: (state: any) => ({ ...state, loading: true }),
+    txHistorySuccess: (state: any, action) => ({
+      ...state,
+      loading: false,
+      records: action.payload,
+    }),
+    txHistoryFail: (state: any, action) => ({
+      ...state,
+      loading: false,
+      error: action.payload,
+    }),
+  },
+});
+
+// TODO - make thunk methods consistent (eg. use createAsyncChunk or not)
+// leaving like this to compare
+export const fetchAccountTxHistory = (accountId: string) => async (
+  dispatch: AppDispatch,
+) => {
+  dispatch(txHistorySlice.actions.txHistoryPending());
+  const dataProvider = new DataProvider({
+    serverUrl: "https://horizon-testnet.stellar.org",
+    accountOrKey: accountId,
+    networkPassphrase: StellarSdk.Networks.TESTNET,
+  });
+  let payments: Array<Types.Payment> | null = null;
+  try {
+    payments = (await dataProvider.fetchPayments())?.records;
+  } catch (err) {
+    return dispatch(txHistorySlice.actions.txHistoryFail(payments));
+  }
+  return dispatch(txHistorySlice.actions.txHistorySuccess(payments));
+};
+
+export const { reducer } = txHistorySlice;
```

Comment:
I moved the txHistory reducer into its own duck file. I thought it was too coupled with account since it's a totally different api call.

But this also shows an example of async calls using createSlice's **reducers** field instead of **extraReducers**. I don't think we need to use **extraReducers** on the accountSlice ?

Reply:
Good call about moving `txHistory` to its own file. 👍 

I like that when using `extraReducers` there are no manual `dispatch` calls (it's a clean `try catch` with a response). I guess it comes down to where you add extra logic: to the function or reducers. I'm still learning RTK and either approach is OK with me.